src/fr/poseidonscans/src/eu/kanade/tachiyomi/extension/fr/poseidonscans/PoseidonScans.kt

```
@@ -1,14 +1,569 @@
+    private fun extractNextJsPageData(document: Document): JsonObject? {
+        val currentUrl = document.location()
+        val isSeriesPage = currentUrl.contains("/series")
+
+        try {
+            document.selectFirst("script#__NEXT_DATA__")?.data()?.also { scriptData ->
+                try {
+                    val rootJson = json.decodeFromString<JsonObject>(scriptData)
+                    val pageProps = rootJson["props"]?.jsonObject?.get("pageProps")?.jsonObject
+                    if (pageProps != null) {
+                        if (isSeriesPage) {
+                            if (pageProps.containsKey("initialData") || pageProps.containsKey("mangas") || pageProps.containsKey("series")) {
+                                return pageProps
+                            }
+                        } else {
+                            if (pageProps.containsKey("initialData") || pageProps.containsKey("manga") ||
+                                pageProps.containsKey("chapter") || pageProps.containsKey("images")
+                            ) {
+                                return pageProps
+                            }
+                        }
+                    }
+                    rootJson["initialData"]?.jsonObject?.let { initialData ->
+                        if (isSeriesPage) {
+                            if (initialData.containsKey("mangas") || initialData.containsKey("series")) {
+                                return initialData
+                            }
+                        } else {
+                            if (initialData.containsKey("manga") || initialData.containsKey("chapter") || initialData.containsKey("images")) {
+                                return initialData
+                            }
+                        }
+                    }
+                } catch (e: Exception) {
+                    // silently ignore
```

why silently ignore?

---

Yeah my bad im adding a log here